### DISCUSSION TITLE
Combining an intergrated BC

### DISCUSSION THREAD

**Discussion by gj19866** [idaholab/moose#26847] [Q&A General]
Hi there,

I want to create a boundary condition such that $\hat{n} \cdot (\nabla \chi - \boldsymbol{A} )= 0$. 
Is this possible within the integrated BC method? 
I have read the docs but I am slightly confused about it. 

All the best,
Gillian

**Comment by GiudGiud** [comment]
Hello

Is chi the variable?
And depending on the weak form of your equation on the volume, this looks a lot like a NeumannBC for the diffusion equation. 

Guillaume

**Reply by gj19866** [reply]
Hello Guillaume,

Chi is a variable (and is the variable I want the BC to act on). $\boldsymbol{A}$ is a Vector variable however, and hence I thought that I couldn't use the standard NeumannBC for this case?

Thank you,
Gillian

**Reply by GiudGiud** [reply]
Is A a nonlinear variable? Or something that is fixed or known on every time step?

**Reply by gj19866** [reply]
A is defined by a Parsed function, and is known at very time step, and varys only in space and not time

**Reply by GiudGiud** [reply]
If you defined A * n as a function, then you could use a FunctionNeumannBC

**Reply by gj19866** [reply]
Amazing thank you. How could I go about defining that?

**Reply by gj19866** [reply]
The reason being, I want to apply this to non-trivial geometries.

**Reply by GiudGiud** [reply]
Do you plan to use a fixed component for the normal?
Or the actual norm to the surface?
The latter cannot be done in a function. We d have to use an auxKernel to do this

**Reply by gj19866** [reply]
I think I would need the actual norm to the surface. 
Creating an auxVar seems like a reasonable solution, I just don't know what functionality to use to create an auxvariable with value $\hat{n} \cdot A$.

**Reply by gj19866** [reply]
And can that be parsed into the NeumannBC the same as you would a value?

**Reply by GiudGiud** [reply]
so if we use a auxiliary variable, you ll want to use:
https://mooseframework.inl.gov/moose/source/bcs/CoupledVarNeumannBC.html
instead of a FunctionNeumannBC

to create n*A in an auxiliary variable, you can use a CONSTANT MONOMIAL (let's start with that) auxiliary variable then a simple auxkernel that is boundary-restricted.

The n * A operation is somewhat close to the AdvectiveFluxAux operation? You might be able to make a new one?
https://mooseframework.inl.gov/moose/source/auxkernels/AdvectiveFluxAux.html

**Reply by gj19866** [reply]
Thank you very much. I am getting slightly lost looking at the AdvectiveFluxAux.C as these are several different switches to activate. Which one is applicable in this case? Thank you for the help.

**Reply by GiudGiud** [reply]
You ll have to modify it I think.
It s simply calculation, multiplying a vector by the normal

**Reply by gj19866** [reply]
I now have it working.

Thank you very much for all the help Guillaume.

All the best from the UK!

**Reply by GiudGiud** [reply]
Awesome!
